### PR TITLE
Migrate Validation orchestrator jobs to new SDK and Managed Identities

### PR DESCRIPTION
**Pull request by advay26**
Addresses https://github.com/NuGet/Engineering/issues/5440

### Background

This migrates some of the Validation Jobs to use MSI with the new SDKs. I've deployed these changes to DEV using these config files: https://nuget.visualstudio.com/NuGetMicrosoft/_git/NuGetDeployment/pullrequest/2473, and E2E tests pass + I don't see any exceptions in AI logs.

These are the jobs I tested:
- [[NuGet][Continuous] Validation.Orchestrator](https://devdiv.visualstudio.com/DevDiv/_release?_a=releases&view=all&definitionId=1617)
- [[NuGet][Continuous] Validation.Symbols.Job](https://devdiv.visualstudio.com/DevDiv/_release?_a=releases&view=all&definitionId=1584)
- [[NuGet][Continuous] Validation.SymbolsOrchestrator](https://devdiv.visualstudio.com/DevDiv/_release?_a=releases&view=all&definitionId=1605)
- [[NuGet][Daily] Validation.PackageSigning.RevalidateCertificate](https://devdiv.visualstudio.com/DevDiv/_release?_a=releases&view=all&definitionId=1609)

### Description of the changes

I had to configure the Lease service (used by the Validation Orchestrator jobs) to use MSI. All other jobs and storage connections work without any additional code changes. Andrei's changes in `NuGet.Jobs.Common` and `Validation.Common.Job` here addressed them: https://github.com/NuGet/NuGetGallery/pull/10153

`Validation.Common.Job` had a reference for `WindowsAzure.Storage` that I had to remove. `NuGet.Services.V3` is downstream and consumes this package via `Validation.Common.Job`, so I had to add an explicit reference to this package in `NuGet.Services.V3`.

### CODE REVIEW

File: src/NuGet.Services.Validation.Orchestrator/Job.cs [hunk 1]
```diff
@@ -607,5 +607,47 @@ private T GetRequiredService<T>()
         {
             return _serviceProvider.GetRequiredService<T>();
         }
+
+        private static BlobServiceClient CreateBlobServiceClient(
```
Comment (agr) [inline src/NuGet.Services.Validation.Orchestrator/Job.cs:611]: Should we move that to `StorageAccountHelpers` as an extension method for `IComponentContext`? I think @drewgillies could use something similar to this code.

**Review by joelverhagen — APPROVED** [review]

File: src/NuGet.Services.Validation.Orchestrator/Job.cs (hunk 1 above)
Comment (advay26) [reply src/NuGet.Services.Validation.Orchestrator/Job.cs]: Makes sense. I've moved it now

File: src/NuGet.Jobs.Common/StorageAccountExtensions.cs [hunk 2]
```diff
@@ -180,6 +182,48 @@ private static CloudBlobClientWrapper CreateCloudBlobClient(
                 requestTimeout);
         }
 
+        public static BlobServiceClient CreateBlobServiceClient(
+            StorageMsiConfiguration storageMsiConfiguration,
+            string storageConnectionString,
+            TimeSpan? requestTimeout = null)
+        {
+            BlobClientOptions blobClientOptions = new BlobClientOptions();
+            if (requestTimeout.HasValue)
```
Comment (erdembayar) [inline src/NuGet.Jobs.Common/StorageAccountExtensions.cs:191]: Maybe it's too far-fetched. Should we check to see if it's not negative or too high of a value?

File: src/NuGet.Services.V3/NuGet.Services.V3.csproj [hunk 3]
```diff
@@ -7,6 +7,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="WindowsAzure.Storage" />
```
Comment (erdembayar) [inline src/NuGet.Services.V3/NuGet.Services.V3.csproj:10]: Why are we still referencing legacy Azure sdk here?

Comment (advay26) [reply src/NuGet.Services.V3/NuGet.Services.V3.csproj:10]: I was making changes to `Validation.Common.Job`, and it had a direct reference to `WindowsAzure.Storage` that I had to remove. `NuGet.Services.V3` is downstream from this project and consumes the old Azure SDK package via `Validation.Common.Job`, so I had to add an explicit reference to this package in `NuGet.Services.V3`.

I'm guessing once the V3 jobs have been migrated, that can be removed too.

**Review by erdembayar — APPROVED** [review]

**Review by agr — APPROVED** [review]

**Pull request merged**